components/files/escape.cpp

```diff
@@ -27,7 +27,8 @@ namespace Files
     std::string EscapeHashString::processString(const std::string & str)
     {
         std::string temp = boost::replace_all_copy<std::string>(str, std::string() + (char)escape_hash_filter::sEscape + (char)escape_hash_filter::sHashIdentifier, "#");
-        boost::replace_all(temp, std::string() + (char)escape_hash_filter::sEscape + (char)escape_hash_filter::sEscapeIdentifier, std::string((char)escape_hash_filter::sEscape, 1));
+        auto format = std::string(1, (char)escape_hash_filter::sEscape);
+        boost::replace_all(temp, std::string() + (char)escape_hash_filter::sEscape + (char)escape_hash_filter::sEscapeIdentifier, format);
```

Comment:
Can we re-write this to not use boost?  Either with std::replace_copy, or std::replace ?
http://www.martinbroadhurst.com/replacing-all-occurrences-of-a-character-in-a-stdstring.html

Reply:
http://en.cppreference.com/w/cpp/string/basic_string/replace

```
std::string& replaceAll(std::string& context, std::string const& from, std::string const& to)
{
    std::size_t lookHere = 0;
    std::size_t foundHere;
    while((foundHere = context.find(from, lookHere)) != std::string::npos)
    {
          context.replace(foundHere, from.size(), to);
          lookHere = foundHere + to.size();
    }
    return context;
}
```

Reply:
I'll look into removing mPrevious and replacing the boost stuff when I get home. Didn't wan to be too intrusive, so I just applied the minimal fixes.

Reply:
If you can, that would be great. Anything that lessens our dependency on boost.
Something that fixes this instance in this PR.

Then a second PR that then uses the same technique, but everywhere boost::replace_all is used.

Unless @scrawl or @zinnschlag disagrees. ;)

Reply:
I'm all for removing boost, but I won't do it over the whole code base. Should be a separate issue, IMHO.

Reply:
Well... baby-steps. :)

Reply:
Yeah, excatly. Sorry, got wound up in some after-work stuff. ;) Will get to it on wednesday, I hope.